com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadowManagerDirectional.RenderGraph.cs

```diff
@@ -0,0 +1,158 @@
+
+            // Apply the spatial denoiser
+            HDDiffuseShadowDenoiser shadowDenoiser = GetDiffuseShadowDenoiser();
+            TextureHandle denoisedBuffer = shadowDenoiser.DenoiseBufferDirectional(renderGraph, hdCamera,
+                            depthBuffer, normalBuffer,
+                            temporalFilterResult.outputSignal, temporalFilterResult.outputSignalDistance,
+                            m_CurrentSunLightAdditionalLightData.filterSizeTraced, m_CurrentSunLightAdditionalLightData.angularDiameter * 0.5f, !m_CurrentSunLightAdditionalLightData.colorShadow);
+
+            // Now that we have overriden this history, mark is as used by this light
+            hdCamera.PropagateShadowHistory(m_CurrentSunLightAdditionalLightData, dirShadowIndex, GPULightType.Directional);
+
+            return denoisedBuffer;
+        }
+
+        class RTSDirectionalTracePassData
+        {
+            public RTShadowDirectionalTraceParameters parameters;
+            public TextureHandle depthStencilBuffer;
+            public TextureHandle normalBuffer;
+            public TextureHandle directionBuffer;
+            public TextureHandle rayCountTexture;
+            public TextureHandle velocityBuffer;
+            public TextureHandle distanceBuffer;
+            public TextureHandle outputShadowBuffer;
+        }
+
+        void RenderRayTracedDirectionalScreenSpaceShadow(RenderGraph renderGraph, HDCamera hdCamera, TextureHandle depthBuffer, TextureHandle normalBuffer, TextureHandle motionVetorsBuffer, TextureHandle rayCountTexture, TextureHandle screenSpaceShadowArray)
+        {
+            TextureHandle directionalShadow;
+            TextureHandle velocityBuffer;
+            TextureHandle distanceBuffer;
+            RTShadowDirectionalTraceParameters rtsdtParams = PrepareRTShadowDirectionalTraceParameters(hdCamera, m_CurrentSunLightAdditionalLightData);
+            using (var builder = renderGraph.AddRenderPass<RTSDirectionalTracePassData>("Directional RT Shadow", out var passData, ProfilingSampler.Get(HDProfileId.RaytracingDirectionalLightShadow)))
+            {
+                passData.parameters = rtsdtParams;
+
+                // Input Buffer
+                passData.depthStencilBuffer = builder.UseDepthBuffer(depthBuffer, DepthAccess.Read);
+                passData.normalBuffer = builder.ReadTexture(normalBuffer);
+                passData.directionBuffer = builder.CreateTransientTexture(new TextureDesc(Vector2.one, true, true) { colorFormat = GraphicsFormat.R16G16B16A16_SFloat, enableRandomWrite = true, name = "Direction Buffer" });
+
+                // Debug buffers
+                passData.rayCountTexture = builder.ReadTexture(builder.WriteTexture(rayCountTexture));
+
+                // Output Buffers
+                passData.velocityBuffer = builder.ReadTexture(builder.WriteTexture(renderGraph.CreateTexture(new TextureDesc(Vector2.one, true, true)
```

Comment:
Do we actually write back to the velocity buffer or is that a copy/paste mistake?

Reply:
Seems like everything is read/write here so it seems weird to me but maybe that's intended?